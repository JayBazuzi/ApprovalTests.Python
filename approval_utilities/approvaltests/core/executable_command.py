from abc import abstractmethod, ABC


class ExecutableCommand(ABC):
    @abstractmethod
    def get_command(self) -> str:
        raise Exception("Interface member not implemented")

    @abstractmethod
    def execute_command(self, command: str) -> str:
        raise Exception("Interface member not implemented")
