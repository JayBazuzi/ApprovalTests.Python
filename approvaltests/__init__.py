"""Approvaltests module."""
from .version import *
from .utilities import *
from .approvals import *
from .asserts import *

__author__ = "Chris Lucian; Llewellyn Falco; Jim Counts; Tim Preece"
__version__ = version_number
