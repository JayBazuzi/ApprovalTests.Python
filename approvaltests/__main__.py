from .verify import verify_using_commandline_arguments

verify_using_commandline_arguments()