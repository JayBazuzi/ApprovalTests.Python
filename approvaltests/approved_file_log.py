from pathlib import Path

APPROVAL_TESTS_TEMP_DIRECTORY: str = ".approval_tests_temp"


class ApprovedFilesLog:
    @staticmethod
    def clear_log_file() -> None:
        ApprovedFilesLog.get_approved_files_log().write_text("")

    @staticmethod
    def get_approved_files_log() -> Path:
        path = Path("%s/.approved_files.log" % APPROVAL_TESTS_TEMP_DIRECTORY)
        path.parent.mkdir(parents=True, exist_ok=True)

        return path

    @staticmethod
    def log(approved_file):
        with ApprovedFilesLog.get_approved_files_log().open(mode="a") as file:
            file.write(f"{approved_file}\n")


ApprovedFilesLog.clear_log_file()
