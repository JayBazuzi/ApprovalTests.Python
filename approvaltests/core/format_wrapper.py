from typing_extensions import override
import abc
import typing


class FormatWrapper(abc.ABC):
    @abc.abstractmethod
    def wrap(self, data: typing.Any) -> typing.Any:
        pass

    @abc.abstractmethod
    def is_match(self, data: typing.Any) -> bool:
        pass


class AlwaysMatch(FormatWrapper):
    @override
    def wrap(self, data: typing.Any) -> typing.Any:
        return data

    @override
    def is_match(self, data: typing.Any) -> bool:
        return True
