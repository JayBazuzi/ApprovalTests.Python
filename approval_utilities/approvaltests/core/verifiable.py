from abc import ABC, abstractmethod
from approvaltests.core.options import Options

from approval_utilities.approvaltests.core.verify_parameters import VerifyParameters


class Verifiable(ABC):
    @abstractmethod
    def get_verify_parameters(self, options: "Options") -> VerifyParameters:
        pass
