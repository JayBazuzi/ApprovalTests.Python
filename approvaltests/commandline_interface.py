from pathlib import Path

from approvaltests import verify
from build.lib.approvaltests.commandline_interface import parse_arguments
from namer.cli_namer import CliNamer


# simplify the following lines of code from commandline_interface.py

# 1.  Modify so our future-selves would recognize it
# >>> 2.  Make a test
# 3.  Refactor parse_arguments to be part of a class (suggested by Jay)

# NEw code...
# received = args.received or stdin.read()
# Old code...
# received = args.received
# if args.received is None:
#    received = stdin.read()
# return (args.id, received)



def verify_using_commandline_arguments():
    test_id, received = parse_arguments()
    verify_with_id(received, test_id)


def verify_with_id(received, test_id):
    verify(received, namer=CliNamer(test_id=test_id))
    print(f"Test Passed: {test_id}")


if __name__ == "__main__":
    verify_using_commandline_arguments()


def list_approved_files_in_test_directory(test_dir: Path):
    return [file.name for file in test_dir.iterdir() if file.name.endswith(".approved.txt")]
