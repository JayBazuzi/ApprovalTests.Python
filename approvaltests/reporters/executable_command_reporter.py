import pathlib

from approval_utilities.approvaltests.core.executable_command import ExecutableCommand
from approvaltests import Reporter


class ExecutableCommandReporter(Reporter):
    def __init__(self, command: ExecutableCommand, reporter: Reporter):
        self.command = command
        self.reporter = reporter

    def report(self, received_filename: str, approved_filename: str) -> bool:
        # recieved and approved commands are not the same
        # todo run the content of the file against the executable command
        # execute(read(received))

        self.reporter.report(self.execute_result(received_filename), self.execute_result(approved_filename))
        self.reporter.report(received_filename, approved_filename)
        return True

    def execute_result(self, filename):
        path = pathlib.Path(filename)
        command = path.read_text()
        if command:
            result = self.command.execute_command(command)
        else:
            result = ""
        approved_executed_result_file = f"{path.name[:-len(path.suffix)]}.executed_results.txt"
        pathlib.Path(approved_executed_result_file).write_text(result)
        return approved_executed_result_file
