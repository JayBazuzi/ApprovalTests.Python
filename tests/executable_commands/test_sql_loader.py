from approvaltests.reporters.executable_command_reporter import (
    ExecutableCommandReporter,
)
from approval_utilities.approvaltests.core.executable_command import ExecutableCommand
from approvaltests import verify, verify_executable_command
from tests.executable_commands.country_loader import CountryLoader


def test_to_compare_execute_command():
    verify_executable_command(CountryLoader())


def test_to_compare_execute_command_where_we_see_the_failure():
    verify_executable_command(CountryLoader())


def test_result_formatting_for_results_with_indentation():
    executed_command = "select * from foo"

    class DummyExecutableCommand(ExecutableCommand):
        def get_command(self) -> str:
            return None

        def execute_command(self, command: str) -> str:
            return "result of the query\nstuff that breaks indentation"

    result = ExecutableCommandReporter.execute_command_and_format_result(
        executed_command, DummyExecutableCommand()
    )
    verify(result)


def test_result_formatting_for_non_empty_command():
    executed_command = "select * from foo"

    class DummyExecutableCommand(ExecutableCommand):
        def get_command(self) -> str:
            return None

        def execute_command(self, command: str) -> str:
            return "result of the query"

    result = ExecutableCommandReporter.execute_command_and_format_result(
        executed_command, DummyExecutableCommand()
    )
    verify(result)


def test_result_formatting_for_empty():
    # ApprovalTests doesn't try to execute empty commands
    assert "" == ExecutableCommandReporter.execute_command_and_format_result(None, None)
