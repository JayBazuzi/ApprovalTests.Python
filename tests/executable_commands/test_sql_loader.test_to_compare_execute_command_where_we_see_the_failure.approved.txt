SELECT c.* FROM Country c
